enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+physical devices could be untrusted (devices in public network) or might be
+compromised. Enabling IPsec encryption for this tunnel traffic can prevent the
+traffic data from being monitored and manipulated.
+
+This enhancement allows cluster operators to enable traffic to be encapsulated
+by the IPsec protocol between worker nodes.
+
+## Motivation
+
+Encryption services are recommended when traffic is traversing an untrusted
+network. Encryption services may also be required for regulatory or compliance
+reasons.
+
+### Goals
+
+- Provide an option to enable IPsec encryption of all inter-worker node
+tunnel traffic across an entire Kubernetes cluster that has been configured with
+OVN Kubernetes.
+  - Assume that all nodes will be provisioned with certificates that have been
+  signed by the OpenShift Container Platform internal certificate authority
+  (CA).
+  - This configuration option will only be available on Geneve tunnels
+- All components used to implement this must be FIPs-compliant cryptographic
+components.
+- Provide an implementation that will be deployable on both Public and Private
+Cloud infrastructure. This will require that all inter-worker node traffic can
+be routed by the underlay network without reconfiguration of underlay firewall
+policies.
+- Reuse the same certificates, keys and CA used by the Cluster Management TLS
+implementation.
+
+### Non-Goals
+
+- Will not provide any Hardware Offload of the IPsec protocol. Although some
+checksums may be offloaded by standard NICs without any special enablement.
+- Will not provide any implementation for DPDK interfaces.
+- Will not provide an option to enable tunnels that have been configured with
+self-signed certificates or pre-shared keys.
+- Will not do any explicit performance optimization of the datapath. As the
+implementation of this enhancement will primarily require changes to the control
+and management path, it is expected that this will not alter the performance of
+the current OVS/Kernel IPsec implementation.
+- Will not provide any interface to allow a cluster operator to specify
+preferred encryption, integrity, authentication algorithms
+- Will not explicitly encrypt cluster management traffic. Cluster management
+traffic has already been encrypted using TLS.
+- Will not allow granular configuration of which nodes will have IPsec enabled
+(for example Namespace to Namespace encryption) or how IPsec is configured
+between specific nodes. This enhancement will enable IPsec between all worker
+nodes within the cluster indiscriminately.
+- Will not provide IPsec connectivity between any worker node that does not form
+part of the Kubernetes cluster. This enhancement is only for Node-to-Node
+(east-west) communication within the cluster.
+
+## Proposal
+
+The following changes are proposed to the [network.operator.openshift.io API](https://github.com/openshift/api/blob/master/operator/v1/types_network.go).
```

Comment:
I wonder how we can highlight which field is new vs. existing?  Maybe just expand this comment to say that the `EnableIPSec` field is the only new one?

Reply:
Resolved elsewhere.